library/Rules/FactorOf.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of Respect/Validation.
+ *
+ * (c) Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ *
+ * For the full copyright and license information, please view the "LICENSE.md"
+ * file that was distributed with this source code.
+ */
+
+namespace Respect\Validation\Rules;
+
+use Respect\Validation\Exceptions\FactorOfException;
+
+class FactorOf extends AbstractRule
+{
+    public function __construct($dividend)
+    {
+        if (!is_numeric($dividend) || (int) $dividend != $dividend || $dividend == 0) {
+            throw new FactorOfException('Dividend must be an integer greater than 0');
```

Comment:
In this case you have to use `ComponentException` instead. That's because the `FactorOfException` must be used only when you call `assert()` or `check()` methods.